ark/mibi/qc_comp.py

```diff
@@ -1,14 +1,160 @@
+        run_label (str):
+            The label of the run (specified on the user's MIBItracker run page)
+        base_dir (str):
+            Where to place the created `tiff_dir`
+        tiff_dir (str):
+            The name of the data directory in `base_dir` to write the run's image data to
+        xml_name (str):
+            The name of the XML file to save the run metadata to in `data_dir`.
+            Needs to be suffixed by `.xml`
+        img_sub_folder (str):
+            If specified, the subdirectory inside each FOV folder in `data_dir` to place
+            the image data into
+        fovs (list):
+            A list of FOVs to subset over. If `None`, uses all FOVs.
+        channels (lsit):
+            A list of channels to subset over. If `None`, uses all channels.
+    """
+
+    # verify that base_dir provided exists
+    if not os.path.exists(base_dir):
+        raise FileNotFoundError("base_dir %s does not exist" % base_dir)
+
+    # create the MIBItracker request helper
+    mr = create_mibitracker_request_helper(email, password)
+
+    # get the run info using the run_name and the run_label
+    # NOTE: there will only be one entry in the 'results' key with run_name and run_label specified
+    run_info = mr.search_runs(run_name, run_label)
+
+    # extract the name of the FOVs and their associated internal IDs
+    run_fov_names = [img['number'] for img in run_info['results'][0]['imageset']['images']]
+    run_fov_ids = [img['id'] for img in run_info['results'][0]['imageset']['images']]
+
+    # if fovs is None, ensure all of the fovs in run_fov_names_ids are chosen
+    if fovs is None:
+        fovs = run_fov_names
+
+    # ensure all of the fovs are valid (important if the user explicitly specifies fovs)
+    misc_utils.verify_in_list(
+        provided_fovs=fovs,
+        mibitracker_run_fovs=run_fov_names
+    )
+
+    # extract the name of all the channels
+    # NOTE: this set of channels will be the same across all FOVs in the run
+    run_channels = run_info['results'][0]['imageset']['images'][0]['pngs']
+
+    # if channels is None, ensure all of the channels in run_channels are chosen
+    if channels is None:
+        channels = run_channels
+
+    # ensure all of the channels are valid (important if the user explicitly specifies channels)
+    misc_utils.verify_in_list(
+        provided_chans=channels,
+        mibitracker_run_chans=run_channels
+    )
+
+    # download the run metadata
+    run_metadata = mr.download_file(
+        os.path.join(run_info['results'][0]['path'], run_info['results'][0]['xml'])
+    )
+
+    # if the desired tiff_dir exists, remove it
+    if os.path.exists(os.path.join(base_dir, tiff_dir)):
+        rmtree(os.path.join(base_dir, tiff_dir))
+
+    # make the image directory
+    os.mkdir(os.path.join(base_dir, tiff_dir))
+
+    # write the run metadata as XML to specified xml_name in tiff_dir
+    with open(os.path.join(base_dir, tiff_dir, xml_name), 'wb') as outfile:
```

Comment:
What are we using the xml file for?

Reply:
`xml_name` contains the run order, which we'll need to update the QC FOV sorting on the next patch.